VTCareService/src/main/java/edu/vt/cs/vtcare/vtcareservice/services/ProviderService.java

```diff
@@ -28,4 +29,30 @@ public Provider persistProvider(Provider provider) throws SQLException {
         provider.setProviderId(providerId);
         return provider;
     }
+
+    public List<Provider> getProvider(String id) throws Exception{
+        long num_id =Long.parseLong(id);
+        try {
+            return providerDao.getProvider(num_id);
+        }
+        catch(Exception e){
+            System.out.println(e);
+            throw new Exception();
```

Comment:
please don't instantiate redundant exception objects - for explanation please refer other comments.